test/FastTests/Client/Indexing/DebugIndexing.cs

```diff
@@ -25,10 +27,11 @@ private class Person
             public string Name { get; set; }
         }
 
-        [Fact]
-        public async Task QueriesRunning()
+        [Theory]
+        [MemberData(nameof(SearchEngineTypeValue.Data), MemberType= typeof(SearchEngineTypeValue))]
+        public async Task QueriesRunning(string searchEngineType)
         {
-            using (var store = GetDocumentStore())
+            using (var store = GetDocumentStore(new Options(){ModifyDatabaseRecord = d => d.Settings[RavenConfiguration.GetKey(x => x.Indexing.AutoIndexingEngineType)] = searchEngineType}))
```

Comment:
introduce factory 'ForSearchEngine' and instance method 'WithSearchEngine' if needed

Reply:
done